### PR TITLE
Move testing instructions to dev guide

### LINKED ISSUE #305: [Feature]: Make test suite require fewer dependencies
### What would you like to see added to NeuroConv?

It would be cool to make the test suite more accessible.

### Is your feature request related to a problem?

I get numerous test failures, mainly due to the test suite having a large and dependency graph specific to the package, but also due to missing test data.

Here is the [full build and test log](https://ppb.chymera.eu/1415d6.log).

(1) Is there any way to download the test data separeatly? Going by the log, I thought it might be under https://github.com/catalystneuro/ephy_testing_data but there's nothing there

(2) would it be possible for the test suite to just skip tests if optional dependencies are not installed? 

### Do you have any interest in helping implement the feature?

No.

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/catalystneuro/neuroconv/blob/main/.github/CODE_OF_CONDUCT.rst)
- [X] Have you ensured this bug was not already [reported](https://github.com/catalystneuro/neuroconv/issues)?

### PR DESCRIPTION
**Pull request by CodyCBakerPhD**
fix #305

### CODE REVIEW

File: docs/developer_guide/testing_suite.rst [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+There are three broad classes of tests in this regard...
+
+
+
+Minimal
+-------
+
+These test internal functionality using only minimal dependencies or pre-downloaded data.
+
+Subfolders: `tests/test_minimal <https://github.com/catalystneuro/neuroconv/tree/main/tests/test_minimal>`_ and `tests/test_internals <https://github.com/catalystneuro/neuroconv/tree/main/tests/test_internals>`_
```
Comment (CodyCBakerPhD) [inline docs/developer_guide/testing_suite.rst:19]: Using this type of link ensures that our link checker CI will remind us to update the links here if we end up changing the testing folder structure or names in any way

File: docs/developer_guide/testing_suite.rst [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+Subfolders: `tests/test_ophys <https://github.com/catalystneuro/neuroconv/tree/main/tests/test_ophys>`_, `tests/test_ecephys <https://github.com/catalystneuro/neuroconv/tree/main/tests/test_ecephys>`_, `tests/test_behavior <https://github.com/catalystneuro/neuroconv/tree/main/tests/test_behavior>`_, and `tests/test_text <https://github.com/catalystneuro/neuroconv/tree/main/tests/test_text>`_
+
+These can be run in isolation using ``pip install -e neuroconv[test,<modality>]`` and calling ``pytest tests/test_<modality>`` where ``<modality>`` can be any of ``ophys``, ``ecephys``, ``text``, or ``behavior``.
+
+
+
+On Data
+-------
+
+For proprietary formats, we like to be able to regularly test our conversions against small snnippets of real data.
```
Comment (bendichter) [inline docs/developer_guide/testing_suite.rst:39]:
```suggestion
For proprietary formats, we regularly test our conversions against small snnippets of real data.
```

File: docs/developer_guide/testing_suite.rst [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+""""""""""""""""""""""""""
+``datalad install -rg https://gin.g-node.org/NeuralEnsemble/ephy_testing_data``
+
+For optical physiology data
+"""""""""""""""""""""""""""
+``datalad install -rg https://gin.g-node.org/CatalystNeuro/ophys_testing_data``
+
+For behavioral data
+"""""""""""""""""""
+``datalad install -rg https://gin.g-node.org/CatalystNeuro/behavior_testing_data``
```
Comment (bendichter) [inline docs/developer_guide/testing_suite.rst:55]: how do you update the data in these folders?

**Review by bendichter — CHANGES_REQUESTED** [review]
see comments

File: docs/developer_guide/testing_suite.rst (hunk 3 above)
Comment (CodyCBakerPhD) [reply docs/developer_guide/testing_suite.rst:55]: I think `datalad update`

**Review by bendichter — APPROVED** [review]

**Pull request merged**